src/app/space/create/deployments/apps/deployment-card-container.component.spec.ts

```diff
@@ -35,34 +39,83 @@ class FakeDeploymentCardComponent {
 
 describe('DeploymentCardContainer', () => {
   type Context = TestContext<DeploymentCardContainerComponent, HostComponent>;
-
   const environments: string[] = ['envId1', 'envId2'];
+  let mockDeploymentsService: jasmine.SpyObj<DeploymentsService>;
+  let mockContexts: any = jasmine.createSpy('Contexts');
+  let current = {
+    path: 'mock-path',
+    user: {
+      attributes: {
+        username: 'mock-username'
+      }
+    }
+  };
+  mockContexts.current = Observable.of(current);
 
-  initContext(DeploymentCardContainerComponent, HostComponent,
-    {
-      declarations: [FakeDeploymentCardComponent]
-    },
-    (component: DeploymentCardContainerComponent) => {
-      component.spaceId = 'space';
-      component.environments = Observable.of(environments);
-      component.application = 'app';
-    });
+  describe('User has deployed applications', () => {
+    mockDeploymentsService = jasmine.createSpyObj('DeploymentsService', ['hasDeployments']);
+    mockDeploymentsService.hasDeployments.and.returnValue(Observable.of(true));
+    initContext(DeploymentCardContainerComponent, HostComponent,
+      {
+        declarations: [FakeDeploymentCardComponent],
+        providers: [
+          { provide: Contexts, useValue: mockContexts },
+          { provide: DeploymentsService, useValue: mockDeploymentsService }
+        ],
+        schemas: [NO_ERRORS_SCHEMA]
+      },
+      (component: DeploymentCardContainerComponent) => {
+        component.spaceId = 'space';
+        component.environments = Observable.of(environments);
+        component.applications = ['app'];
+        component.collapsed['app'] = true;
```

Comment:
Does this `collapsed` property need to be set here? IIRC this customizer hook gets run before the `ngOnInit` lifecycle hook does (or that was the intention when I added the customizer function), so I think the component should take of initializing its own `collapsed` map property.